test/strategy/DebugConfigStrategyTest.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+var common = require('../common');
+var assert = common.assert;
+
+var strategy = require('../../lib/strategy');
+var DebugConfigStrategy = strategy.DebugConfigStrategy;
+
+function MockLogger(listener) {
+  this.listener = listener;
+}
+
+MockLogger.prototype.debug = function (message) {
+  this.listener('debug', message);
+};
+
+describe('DebugConfigStrategy', function () {
+  var mockLogger;
+  var testSection;
+  var testConfig;
+  var testStrategy;
+  var testListener;
+  var loggedTestMessages = [];
+
+  before(function () {
+    testSection = 'test';
```

Comment:
I always use hardcoded uuids instead of these dummy strings, it helped me a lot when trying to debug tests that failed and tests that were *leaking*  into other tests. Sure, doesn't look to do much at this point, but later it might :)